R/tt_compatibility.R

```diff
@@ -374,6 +374,25 @@ rbindl_rtables <- function(x, gap = 0, check_headers = TRUE) {
 
     lapply(x, function(xi) chk_compat_cinfos(x[[1]], xi)) ##col_info(xi)))
 
+    rbind_annot <- list(title = "", subtitles = character(), footer = character(), pf = character())
+    # Titles/footer info are (independently) retained from first object if all other objects have none
+    all_titles <- lapply(x, function(x) if (is.null(attr(x, "main_title"))) "" else main_title(x))
```

Comment:
If `main_title()` is used instead of `attr()` then an error occurs when binding with non-tables (`rrow()` objects, etc.) which don't have these attributes.

Reply:
The fact that this happens to work for rtables objects is because slots are stored in an S4 object's attributes. We dont' want to use attr here.

There are a couple of to get around this that doesn't violate the abstraction.

The easiest is probably to do 

```
setMethod("main_title", "TableRow", function(x) "")
```

(and the same for the other title/footer getters). Then we don't even need the if statement.